src/middlewared/middlewared/plugins/activedirectory.py

```diff
@@ -905,12 +1010,44 @@ def validate_credentials(self, ad=None):
                 raise CallError(
                     f'kinit with principal {ad["kerberos_principal"]} failed with error {kinit.stderr.decode()}'
                 )
+        else:
+            if not self.middleware.call_sync('kerberos.realm.query', [('realm', '=', ad['domainname'])]):
+                self.middleware.call_sync(
+                    'datastore.insert',
+                    'directoryservice.kerberosrealm',
+                    {'krb_realm': ad['domainname'].upper()}
+                )
+            self.middleware.call_sync('etc.generate', 'kerberos')
+            kinit = subprocess.run([
+                '/usr/bin/kinit',
+                '--renewable',
+                '--password-file=STDIN',
+                f'{ad["bindname"]}@{ad["domainname"]}'],
+                input=ad['bindpw'].encode(),
+                capture_output=True
+            )
+            if kinit.returncode != 0:
+                realm = self.middleware.call_sync(
+                    'kerberos.realm.query',
+                    [('realm', '=', ad['domainname'])],
+                    {'get': True}
+                )
+                self.middleware.call_sync('kerberos.realm.delete', realm['id'])
+                raise CallError(f"kinit for domain [{ad['domainname']}] with password failed: {kinit.sderr.decode()}")
```

Comment:
I believe this should be `stderr` ?

Reply:
Indeed. Thank you.

Reply:
Indeed, it should. Thank you for catching that.